app/services/foreman_ansible/ansible_report_importer.rb

```diff
@@ -15,6 +15,19 @@ def host
           @host = Nic::Interface.find_by(:ip => hostname).host
         end
         super
+        partial_hostname_match(hostname)
+      end
+
+      def partial_hostname_match(hostname)
+        return @host unless @host.new_record?
+        hosts = Host.where(Host.arel_table[:name].matches("%#{hostname}%"))
```

Comment:
this should be matching `hostname.%` and perhaps start with short name exact match first